src/components/TimeQuerier/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import {
+  Text,
+  VStack,
+  HStack,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper,
+  useStyleConfig,
+  Box,
+  Button,
+  Alert,
+  AlertIcon,
+} from "@chakra-ui/react";
+import moment from "moment";
+import Datetime from "react-datetime";
+import constants from "../../utils/constants";
+import { TimeQuerierStore } from "../../store/timeQuerier";
+
+const TimeQuerier: React.SFC = () => {
+  const styles = useStyleConfig("DateTime", {});
+  const {
+    timeQuerierState,
+    changeTimeQuerier,
+  } = TimeQuerierStore.useContainer();
+  const { defaultStepValue, minStepValue, dateFormat, timeFormat } = constants;
+  const [startTime, setStartTime] = useState(
+    moment(timeQuerierState.selectedStartTimestamp)
+  );
+  const [endTime, setEndTime] = useState(
+    moment(timeQuerierState.selectedEndTimestamp)
+  );
+  const [stepTime, setStepTIme] = useState(defaultStepValue);
+  const [error, setError] = useState<undefined | string>(undefined);
+
+  const handleStartChange = (date: moment.Moment | string) => {
+    console.log(date);
+    if (typeof date !== "string") {
+      setStartTime(date);
+      setError(undefined);
+    }
+  };
+  const handleEndChange = (date: moment.Moment | string) => {
+    if (typeof date !== "string") {
+      setEndTime(date);
+      setError(undefined);
+    }
+  };
+  const handleStepChange = (valueAsString: string, valueAsNumber: number) => {
+    if (valueAsString !== "" && valueAsNumber >= minStepValue)
+      setStepTIme(valueAsNumber);
+  };
+  const valid = (current: moment.Moment) => {
+    return current.isBefore(moment());
+  };
+  const handleFetch = () => {
+    if (
+      endTime.isBefore(startTime) ||
+      endTime.isAfter(moment()) ||
+      stepTime < minStepValue
+    )
+      setError("Kindy check you input");
+    else {
+      changeTimeQuerier(
+        startTime.toISOString(),
+        endTime.toISOString(),
+        stepTime
+      );
+    }
+  };
+
+  return (
+    <VStack mt={4} w="100%">
+      <HStack pb="1" w="100%" justifyContent="space-between">
```

Comment:
go with a `Box d="flex"` here with same props

Reply:
Any particular reason for not using Stacks that are just extension of Box

Reply:
Here also is it the same reason? So shall I change HStack here?

Reply:
Yep, just do not use VStacks and HStacks together as much  as possible. Using one instance at a time should be perfectly fine